lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -855,53 +855,35 @@ public static function get_resolved_theme_uris( $theme_json ) {
 		// Using the same file convention when registering web fonts. See: WP_Font_Face_Resolver:: to_theme_file_uri.
 		$placeholder = 'file:./';
 
-		// Top level styles.
-		$background_image_url = $theme_json_data['styles']['background']['backgroundImage']['url'] ?? null;
-		if (
-			isset( $background_image_url ) &&
-			is_string( $background_image_url ) &&
-			// Skip if the src doesn't start with the placeholder, as there's nothing to replace.
-			str_starts_with( $background_image_url, $placeholder ) ) {
+		/*
+		 * Style values are merged at the leaf level, however
+		 * some values provide exceptions, namely style values that are
+		 * objects and represent unique definitions for the style.
+		 */
+		$style_nodes = $theme_json->get_styles_nodes_paths();
```

Comment:
This change can be a separate PR.

Reply:
Anything that generally makes theme.json class related PRs easier is fine by me 😅 

In this case though the PR isn't huge, so I don't mind reviewing as is.